packages/flutter/lib/src/material/time.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show hashValues;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_localizations.dart';
+
+/// Whether the [TimeOfDay] is before or after noon.
+enum DayPeriod {
+  /// Ante meridiem (before noon).
+  am,
+
+  /// Post meridiem (after noon).
+  pm,
+}
+
+/// A value representing a time during the day.
```

Comment:
It would be helpful to add a "See also" section that links to the widgets that use this type.

It might not be such a bad idea to include a sentence or two about why this type exists.